central/graphql/resolvers/node_components.go

```diff
@@ -313,10 +340,71 @@ func (resolver *nodeComponentResolver) TopNodeVulnerability(ctx context.Context)
 		return res, nil
 	}
 
+	// Short path. Full node is embedded when node scan resolver is called.
+	if embeddedComponent := embeddedobjs.NodeComponentFromContext(resolver.ctx); embeddedComponent != nil {
+		var topVuln *storage.NodeVulnerability
+		for _, vuln := range embeddedComponent.GetVulnerabilities() {
+			if topVuln == nil || vuln.GetCvss() > topVuln.GetCvss() {
+				topVuln = vuln
+			}
+		}
+		if topVuln == nil {
+			return nil, nil
+		}
+		return resolver.root.wrapNodeCVE(
+			cveConverter.NodeVulnerabilityToNodeCVE(embeddedobjs.OSFromContext(resolver.ctx), topVuln), true, nil,
+		)
+	}
+
 	return resolver.root.TopNodeVulnerability(resolver.withNodeComponentScope(ctx), RawQuery{})
 }
 
 // UnusedVarSink represents a query sink
 func (resolver *nodeComponentResolver) UnusedVarSink(_ context.Context, _ RawQuery) *int32 {
 	return nil
 }
+
+func getNodeCVEResolvers(ctx context.Context, root *Resolver, os string, vulns []*storage.NodeVulnerability, query *v1.Query) ([]NodeVulnerabilityResolver, error) {
+	query, _ = search.FilterQueryWithMap(query, mappings.NodeVulnerabilityOptionsMap)
+	predicate, err := nodeVulnerabilityPredicateFactory.GeneratePredicate(query)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use the nodes to map CVEs to the nodes and components.
+	idToVals := make(map[string]*nodeCVEResolver)
+	for _, vuln := range vulns {
+		if !predicate.Matches(vuln) {
+			continue
+		}
+		id := cve.ID(vuln.GetCveBaseInfo().GetCve(), os)
+		if _, exists := idToVals[id]; !exists {
+			converted := cveConverter.NodeVulnerabilityToNodeCVE(os, vuln)
+			resolver, err := root.wrapNodeCVE(converted, true, nil)
+			if err != nil {
+				return nil, err
+			}
+			resolver.ctx = embeddedobjs.NodeVulnContext(ctx, vuln)
+			idToVals[id] = resolver
+		}
+	}
+
+	// For now, sort by ID.
+	resolvers := make([]*nodeCVEResolver, 0, len(idToVals))
```

Comment:
nit:  I think naming these so there is more distinction between resolvers and resolverI would be useful.  Also resolverI should probably be plural.